### PR TITLE
Implement ClickHouse storage and refactor data processing

### PR DESCRIPTION
**Pull request by iuwqyir**
### TL;DR

Fully implemented ClickHouse storage and updated block polling logic.

### What changed?

- Replaced in-memory storage with ClickHouse for main and staging databases
- Updated `Commiter` and `Poller` to use a configurable `POLL_FROM_BLOCK` value
- Modified ClickHouse table schemas and queries for blocks, transactions, and logs
- Implemented soft delete functionality for blocks, transactions, and logs
- Updated serialization logic for blocks, transactions, and traces

### How to test?

1. Set up a ClickHouse database and create the necessary tables using the provided SQL scripts
2. Configure the `POLL_FROM_BLOCK` environment variable
3. Run the application and verify that data is being stored in ClickHouse
4. Check that block polling starts from the specified `POLL_FROM_BLOCK` value
5. Verify that soft deletes are working correctly for blocks, transactions, and logs

### CODE REVIEW

**Comment by iuwqyir** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/33?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#33** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/33?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#32** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iuwqyir and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by AmineAfia — APPROVED** [review]

**Review by AmineAfia — APPROVED** [review]
